daemon/daemon.go

```diff
@@ -250,15 +250,9 @@ func (d *Daemon) Dying() <-chan struct{} {
 	return d.tomb.Dying()
 }
 
-func (d *Daemon) auther(r *http.Request) (store.Authenticator, error) {
+func (d *Daemon) auther(user *auth.UserState) (store.Authenticator, error) {
```

Comment:
Yes, the idea definitely wasn't to have a DeviceState inside a UserState.

Anyway, what @matiasb is working on should simplify your life without requiring either of those approaches. The store will be able to ask for device data when it needs it.